pkg/messaging/pair/client.go

```diff
@@ -0,0 +1,91 @@
+package pair
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"log"
+
+	"go.nanomsg.org/mangos/v3/protocol"
+	"go.nanomsg.org/mangos/v3/protocol/pair"
+)
+
+func StartMessagingPairClient(ctx context.Context, nanomsgURL string, requestPair chan RequestPair, responsePair chan ResponsePair) error {
+	socket, err := pair.NewSocket()
+	if err != nil {
+		log.Printf("failed to get new pair socket: %v", err)
+		return err
+	}
+
+	defer func(socketPair protocol.Socket) {
+		if err := socketPair.Close(); err != nil {
+			log.Printf("Failed to close pair socket: %v", err)
+		}
+	}(socket)
+
+	if err := socket.Dial(nanomsgURL); err != nil {
+		log.Printf("failed to dial on pair socket: %v", err)
+		return err
+	}
+
+	go func() {
```

Comment:
Why do you need this goroutine? I think it can be removed. (You can just use for cycle and break if when `<-ctx.Done()`). This issue can be also applied to `pkg/messaging/pubsub/client.go`.

Reply:
Well, the `break` statement wouldn't help to exit from the cycle. It would exit from `select` statement only